fs/fs.go

```diff
@@ -81,13 +82,66 @@ var (
 	fusermountBin               = "fusermount"
 )
 
+type layerQueue struct {
+	mu       sync.Mutex
```

Comment:
I think we can simplify this structure if we replace the mutex + hashset with a `sync.Map` and max size + cur size with a `semaphore.Weighted`.

That way the structures themselves handle concurrent access and we don't need to pull in a whole data structures library.